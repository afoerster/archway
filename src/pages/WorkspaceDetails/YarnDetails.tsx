import * as React from 'react';
import { HiveTable, YarnApplication } from '../../types/Workspace';
import { List } from 'antd';

interface Props {
  poolName: string;
  applications?: YarnApplication[];
}

const renderApplication = (yarnApplication: YarnApplication) => (
  <div>{yarnApplication.name}</div>
);

const YarnDetails = ({ poolName, applications }: Props) => (
  <List
    dataSource={applications}
    renderItem={renderApplication} />
);

export default YarnDetails;
