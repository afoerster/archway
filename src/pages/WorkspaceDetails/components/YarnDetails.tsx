import { Card, Row, Icon } from 'antd';
import * as React from 'react';
import { YarnService } from '../../../types/Cluster';
import { ResourcePoolsInfo, YarnApplication } from '../../../types/Workspace';
import CardHeader from './CardHeader';

interface Props {
  yarn?: YarnService;
  poolName: string;
  pools?: ResourcePoolsInfo;
  onRefreshPools?: () => void;
}

const renderApplication = (yarnUrl?: string) => (yarnApplication: YarnApplication) => (
  <div style={{ textAlign: 'center' }}>
    <div>
      {`${yarnApplication.name}`}
    </div>
    <a
      style={{ fontSize: 12 }}
      target="_blank"
      href={`${yarnUrl}/cluster/app/${yarnApplication.id}/`}>
      DETAILS
    </a>
  </div>
);

const YarnDetails = ({ yarn, poolName, pools, onRefreshPools }: Props) => {
  const rmURL = yarn && `${yarn.resource_manager[0].host}:${yarn.resource_manager[0].port}`;
  return (
    <Card
      actions={[
        <a
          target="_blank"
          href={yarn ? `//${rmURL}` : undefined}>
          Open Resource Manager
        </a>,
      ]}>
      <CardHeader
        icon="rocket"
        heading="Yarn Applications"
        subheading={poolName}
        rightAction={(
          <Icon
            style={{ cursor: 'pointer', alignSelf: 'flex-start' }}
            type="sync"
            spin={pools && pools.loading}
            onClick={() => {
              if ((!pools || !pools.loading) && onRefreshPools) {
                onRefreshPools();
              }
            }}
          />
        )}
      />
      <Row gutter={12} type="flex" justify="center" style={{ marginTop: 18 }}>
        {pools && pools.data && pools.data[0].applications.length > 0
          && pools.data[0].applications.map(renderApplication(rmURL))}
        {(!pools || !pools.data || pools.data.length <= 0
          || pools.data[0].applications.length <= 0) && (
          <div style={{ color: 'rgba(0, 0, 0, .65)' }}>
            No running YARN applications.
          </div>
        )}
      </Row>
    </Card>
  );
};

export default YarnDetails;
