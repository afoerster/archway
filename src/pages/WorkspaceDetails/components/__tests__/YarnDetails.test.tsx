import * as React from 'react';
import { shallow } from 'enzyme';

import YarnDetails from '../YarnDetails';

describe('YarnDetails', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<YarnDetails poolName="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
