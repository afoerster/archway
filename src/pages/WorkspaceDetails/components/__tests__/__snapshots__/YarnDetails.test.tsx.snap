// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YarnDetails renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <YarnDetails
    poolName=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "actions": Array [
        <a
          href="#"
        >
          Open Resource Manger
        </a>,
      ],
      "children": Array [
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "space-between",
            }
          }
        >
          <Label
            style={
              Object {
                "lineHeight": "22px",
              }
            }
          >
            <Icon
              style={
                Object {
                  "fontSize": 22,
                  "paddingRight": 5,
                }
              }
              theme="twoTone"
              twoToneColor="rgb(67, 170, 139)"
              type="rocket"
            />
            Yarn
          </Label>
          <Label
            style={
              Object {
                "fontSize": 10,
                "lineHeight": "18px",
              }
            }
          >
            
          </Label>
        </div>,
        <List
          bordered={false}
          dataSource={Array []}
          grid={
            Object {
              "column": 3,
              "gutter": 12,
            }
          }
          loading={false}
          locale={
            Object {
              "emptyText": "No running applications",
            }
          }
          pagination={false}
          prefixCls="ant-list"
          renderItem={[Function]}
          split={true}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Label
              style={
                Object {
                  "lineHeight": "22px",
                }
              }
            >
              <Icon
                style={
                  Object {
                    "fontSize": 22,
                    "paddingRight": 5,
                  }
                }
                theme="twoTone"
                twoToneColor="rgb(67, 170, 139)"
                type="rocket"
              />
              Yarn
            </Label>,
            <Label
              style={
                Object {
                  "fontSize": 10,
                  "lineHeight": "18px",
                }
              }
            >
              
            </Label>,
          ],
          "style": Object {
            "display": "flex",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Icon
                  style={
                    Object {
                      "fontSize": 22,
                      "paddingRight": 5,
                    }
                  }
                  theme="twoTone"
                  twoToneColor="rgb(67, 170, 139)"
                  type="rocket"
                />,
                "Yarn",
              ],
              "style": Object {
                "lineHeight": "22px",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "style": Object {
                    "fontSize": 22,
                    "paddingRight": 5,
                  },
                  "theme": "twoTone",
                  "twoToneColor": "rgb(67, 170, 139)",
                  "type": "rocket",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "Yarn",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
              "style": Object {
                "fontSize": 10,
                "lineHeight": "18px",
              },
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bordered": false,
          "dataSource": Array [],
          "grid": Object {
            "column": 3,
            "gutter": 12,
          },
          "loading": false,
          "locale": Object {
            "emptyText": "No running applications",
          },
          "pagination": false,
          "prefixCls": "ant-list",
          "renderItem": [Function],
          "split": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <a
            href="#"
          >
            Open Resource Manger
          </a>,
        ],
        "children": Array [
          <div
            style={
              Object {
                "display": "flex",
                "justifyContent": "space-between",
              }
            }
          >
            <Label
              style={
                Object {
                  "lineHeight": "22px",
                }
              }
            >
              <Icon
                style={
                  Object {
                    "fontSize": 22,
                    "paddingRight": 5,
                  }
                }
                theme="twoTone"
                twoToneColor="rgb(67, 170, 139)"
                type="rocket"
              />
              Yarn
            </Label>
            <Label
              style={
                Object {
                  "fontSize": 10,
                  "lineHeight": "18px",
                }
              }
            >
              
            </Label>
          </div>,
          <List
            bordered={false}
            dataSource={Array []}
            grid={
              Object {
                "column": 3,
                "gutter": 12,
              }
            }
            loading={false}
            locale={
              Object {
                "emptyText": "No running applications",
              }
            }
            pagination={false}
            prefixCls="ant-list"
            renderItem={[Function]}
            split={true}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Label
                style={
                  Object {
                    "lineHeight": "22px",
                  }
                }
              >
                <Icon
                  style={
                    Object {
                      "fontSize": 22,
                      "paddingRight": 5,
                    }
                  }
                  theme="twoTone"
                  twoToneColor="rgb(67, 170, 139)"
                  type="rocket"
                />
                Yarn
              </Label>,
              <Label
                style={
                  Object {
                    "fontSize": 10,
                    "lineHeight": "18px",
                  }
                }
              >
                
              </Label>,
            ],
            "style": Object {
              "display": "flex",
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Icon
                    style={
                      Object {
                        "fontSize": 22,
                        "paddingRight": 5,
                      }
                    }
                    theme="twoTone"
                    twoToneColor="rgb(67, 170, 139)"
                    type="rocket"
                  />,
                  "Yarn",
                ],
                "style": Object {
                  "lineHeight": "22px",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "style": Object {
                      "fontSize": 22,
                      "paddingRight": 5,
                    },
                    "theme": "twoTone",
                    "twoToneColor": "rgb(67, 170, 139)",
                    "type": "rocket",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Yarn",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "",
                "style": Object {
                  "fontSize": 10,
                  "lineHeight": "18px",
                },
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": false,
            "dataSource": Array [],
            "grid": Object {
              "column": 3,
              "gutter": 12,
            },
            "loading": false,
            "locale": Object {
              "emptyText": "No running applications",
            },
            "pagination": false,
            "prefixCls": "ant-list",
            "renderItem": [Function],
            "split": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
